doc/BuildingWithCMake.md

```diff
@@ -54,6 +54,104 @@ CMake can generate build files for a number of backends. You can select the gene
 available generators with `--help`. We recommend using with the `Ninja` generator, when available. More information is
 available [here](https://cmake.org/cmake/help/latest/manual/cmake-generators.7.html).
 
+## Configuring the build on Windows
+
+There are three widely used C/C++ toolchains for Windows: *MSYS2/MinGW*, *Microsoft Visual Studio* and *LLVM/Clang*.
+Let's consider how to use each toolchain to configure and build `OMR` with `CMake`.
+
+### Common prerequisite
+
+*Perl* must be installed otherwise `CMake` won't be available to configure the project:
+
+```bash
+-- Could NOT find Perl (missing: PERL_EXECUTABLE)
+CMake Error at cmake/modules/OmrCompilerSupport.cmake:42 (message):
+  Perl not found
+```
+
+[Strawberry Perl](http://strawberryperl.com/) is absolutelly suitable for `CMake` on Windows, please do not forget
+to add the path to `perl.exe` (for example `C:\StrawberryPerl\perl\bin`) to the `PATH` environment variable.
+
+### Configuring the build for using MSYS2/MinGW
+
+The code of `OMR` uses the Microsoft extension to `C` for supporting *Windows Structured Exception Handling
+([SEH](https://msdn.microsoft.com/ru-ru/library/windows/desktop/ms680657(v=vs.85).aspx))*. *MinGW* haven't supported it
+for a long time. Some workarounds exist but none of them is suitable to use in production-quality software.
+
+### Configuring the build for using Visual Studio Build Tools
+
+*Visual Studio Build Tools* is a bundle with a standalone compiler, libraries and scripts to build C++ libraries and
+applications targeting Windows desktop. The [build tools](http://landinghub.visualstudio.com/visual-cpp-build-tools)
+are available for download on Microsoft's website.
+
+The following components of the build tools are recommended to be installed:
+
+ * Windows 10 SDK (>= 10.0.16299.0) for Desktop C++
+ * Visual C++ tools for CMake
+
+**Note:** 
+To use the `NMake` or `Ninja` generator with *Visual C++*, `cmake` must be run from a shell that can use 
+the compiler `cl` from the command line. To enable the shell, please run
```

Comment:
These instructions aren't really special to OMR, and pretty specific to the install location and version of VS. Can we just link to [this](https://docs.microsoft.com/en-us/cpp/build/building-on-the-command-line) broader guide on setting up a windows machine for command-line development?

Reply:
@rwy0717 I believe that to have a quote in the document is better than just to give a link.

Reply:
How about both? Even with an inline explanation, I'd still like to link to a more complete resource.

Reply:
Ok, no problem, the link has been added.